src/main/java/org/broadinstitute/hellbender/utils/io/ListFileUtils.java

```diff
@@ -0,0 +1,172 @@
+     * @return entries from values or the files listed in values, filtered by filters
+     */
+    public static Set<String> includeMatching(Collection<String> values, Collection<String> filters, boolean exactMatch) {
+        return includeMatching(values, IDENTITY_STRING_CONVERTER, filters, exactMatch);
+    }
+
+    /**
+     * Converts a type T to a String representation.
+     *
+     * @param <T> Type to convert to a String.
+     */
+    public static interface StringConverter<T> {
+        String convert(T value);
+    }
+
+    /**
+     * Returns a new set of values including only values matching filters
+     * <p/>
+     * Filters may also be a file of filters.
+     * <p/>
+     * The converter should convert T to a unique String for each value in the set.
+     *
+     * @param values     Values or files with values
+     * @param converter  Converts values to strings
+     * @param filters    Filters or files with filters
+     * @param exactMatch If true match filters exactly, otherwise use as both exact and regular expressions
+     * @return entries from values including only values matching filters
+     */
+    public static <T> Set<T> includeMatching(Collection<T> values, StringConverter<T> converter, Collection<String> filters, boolean exactMatch) {
+        Utils.nonNull(values);
+        Utils.nonNull(converter);
+        Utils.nonNull(filters);
+
+        final Set<String> unpackedFilters = unpackSet(filters);
+        final Set<T> filteredValues = new LinkedHashSet<T>();
+        Collection<Pattern> patterns = null;
+        if (!exactMatch) {
+            patterns = compilePatterns(unpackedFilters);
+        }
+        for (T value : values) {
+            String converted = converter.convert(value);
+            if (unpackedFilters.contains(converted)) {
+                filteredValues.add(value);
+            } else if (!exactMatch) {
+                for (Pattern pattern : patterns) {
+                    if (pattern.matcher(converted).find()) {
+                        filteredValues.add(value);
+                    }
+                }
+            }
+        }
+        return filteredValues;
+    }
+
+    /**
+     * Returns a new set of values excluding any values matching filters.
+     * <p/>
+     * Filters may also be a file of filters.
+     * <p/>
+     * The converter should convert T to a unique String for each value in the set.
+     *
+     * @param values     Values or files with values
+     * @param converter  Converts values to strings
+     * @param filters    Filters or files with filters
+     * @param exactMatch If true match filters exactly, otherwise use as both exact and regular expressions
+     * @return entries from values exluding any values matching filters
+     */
+    public static <T> Set<T> excludeMatching(Collection<T> values, StringConverter<T> converter, Collection<String> filters, boolean exactMatch) {
```

Comment:
`includeMatching()` and `excludeMatching()` are almost identical apart from `add()` vs `remove()` -- recommend refactoring so that they call into a shared method.